mls-rs/src/group/mod.rs

```diff
@@ -4729,6 +4789,419 @@ mod tests {
         assert!(all_members_are_in);
     }
 
+    #[cfg(all(
+        feature = "by_ref_proposal",
+        feature = "custom_proposal",
+        feature = "self_remove_proposal"
+    ))]
+    #[maybe_async::test(not(mls_build_async), async(mls_build_async, crate::futures_test))]
+    async fn client_cannot_propose_self_remove_twice() {
+        let mut alice = TestClientBuilder::new_for_test()
+            .with_random_signing_identity("alice", TEST_CIPHER_SUITE)
+            .await
+            .custom_proposal_type(ProposalType::SELF_REMOVE)
+            .build()
+            .create_group(ExtensionList::new(), Default::default())
+            .await
+            .unwrap();
+
+        alice.propose_self_remove(Vec::new()).await.unwrap();
+        let again = alice.propose_self_remove(Vec::new()).await;
+        assert_matches!(again, Err(MlsError::SelfRemoveAlreadyProposed));
+    }
+
+    #[cfg(all(
+        feature = "by_ref_proposal",
+        feature = "custom_proposal",
+        feature = "self_remove_proposal"
+    ))]
+    #[maybe_async::test(not(mls_build_async), async(mls_build_async, crate::futures_test))]
+    async fn client_can_self_remove_and_another_client_can_commit() {
+        // Alice creates a group that supports the self-remove proposal.
+        let mut alice = TestClientBuilder::new_for_test()
+            .with_random_signing_identity("alice", TEST_CIPHER_SUITE)
+            .await
+            .custom_proposal_type(ProposalType::SELF_REMOVE)
+            .build()
+            .create_group(ExtensionList::new(), Default::default())
+            .await
+            .unwrap();
+
+        let (bob_signing_identity, bob_secret_key) =
+            get_test_signing_identity(TEST_CIPHER_SUITE, b"bob").await;
```

Comment:
[nit] Can `self_remove_group_setup` be used?

Reply:
Yes; done!